### PR TITLE
feat: add /review slash command to re-trigger AI code reviews

### PR DESCRIPTION
**Pull request by NickBorgers**
## Summary
- Adds `/review` as a PR comment trigger for the AI Code Review pipeline
- When a collaborator comments `/review` on a PR, the full review cycle re-runs
- Uses `has_plain_token.py` to validate the command is plain text (not in code blocks)
- Resets `agent-reviews-passed` label to allow fresh review cycle

Part of cross-repo pipeline alignment: standardizing `/autoresolve` (issues) + `/review` (PRs) across home-automation, home-automation-plus-security, and hide-my-list.

## Test plan
- [ ] Comment `/review` on a PR and verify the AI Code Review workflow triggers
- [ ] Verify `/review` inside a code block does NOT trigger
- [ ] Verify existing triggers (workflow_run, reopened, workflow_dispatch) still work
- [ ] Verify non-collaborator `/review` comments are rejected

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by NickBorgers** [comment]
## Documentation Review

### Blocking Issues

**`docs/operations/AI_GHA_PIPELINES.md` needs two updates:**

1. **Triggers section** (around line 167–171): Add the new `issue_comment` trigger:
   ```
   - **issue_comment**: When a PR comment contains a plain-text `/review` token (collaborators/owners/members only)
   ```

2. **Review Skip Mechanism section** (around line 248–256): The third bullet only mentions PR reopens. Add:
   ```
   3. When a PR is **reopened** or the **/review command** is used, the label is removed to allow re-review
   ```

   Also worth adding a short paragraph explaining the `/review` command similar to how `/autoresolve` is documented in section 1 — who can use it, what it does, and the security checks (PR-only, plain-text token via `has_plain_token.py`, author association check).

### Conclusion
⚠️ Needs changes

**Comment by NickBorgers** [comment]
## Design Review

Scope check: PASS — 109 additions in one workflow file, proportionate to the stated feature.

### Blocking Issues

**`run_id` fallback is semantically wrong.** When `gh run list --workflow="pr-tests.yml"` returns no results, the script falls back to `echo "${{ github.run_id }}"` — which is the *ai-code-review* workflow's run ID, not a pr-tests run ID. If `tests_passed` happens to resolve `true` at the same time (check-runs API found success but run list found nothing — unlikely but possible on network hiccup or branch mismatch), downstream reviewer jobs get `tests_passed=true` with the wrong `run_id` and will silently pull artifacts from the wrong run or fail artifact download in a confusing way. Fix: use `echo ""` as the fallback. Empty string is an honest sentinel; callers already handle absent run_id for `pull_request` events.

### Worth Considering

- `[ -z "${{ github.event.issue.pull_request }}" ]` relies on GHA rendering an absent object field as empty string — works, but `${{ github.event.issue.pull_request.url != '' }}` would be more explicit about the intent.
- If a `/review` comment is posted on a plain issue (not a PR), the context job continues past the early-exit with `pr_number=''` — verify that downstream reviewer jobs have a guard on empty `pr_number`, or the job will error on variable-empty API calls rather than silently no-op.

### Conclusion
⚠️ Needs changes — one fix needed on the `run_id` fallback before merge.

**Review by NickBorgers — COMMENTED** [review]
Inline observations — see summary comment for full verdict.

File: .github/workflows/ai-code-review.yml [hunk 1]
```diff
@@ -71,14 +75,15 @@ jobs:
       head_repo: ${{ steps.get-pr.outputs.head_repo }}
       # Draft PR status - used to skip auto-fix pipeline for TDD workflows
       is_draft: ${{ steps.get-pr.outputs.is_draft }}
-      # Handle workflow_run, workflow_dispatch, and pull_request events
+      # Handle workflow_run, workflow_dispatch, pull_request, and issue_comment events
```
Comment (NickBorgers) [inline .github/workflows/ai-code-review.yml:81]: `${{ github.event.issue.pull_request }}` is an **object**, and GitHub Actions renders top-level object expressions as an empty string in `run:` contexts. This means the `-z` check is always true — every `issue_comment` event exits here with `valid=false`, so the `/review` command never fires on PR comments.

Use a scalar field instead:
```bash
if [ -z "${{ github.event.issue.pull_request.url }}" ]; then
```
`pull_request.url` is a non-empty string for PR comments and an empty/null string for plain issue comments, making it safe for this check.

**Comment by NickBorgers** [comment]
## Code Review

### Blocking Issues

**No job-level guard for `issue_comment` events** (`.github/workflows/ai-code-review.yml:65`):

Every comment on every issue and PR in the repo now triggers the `get-context` job — checkout, API calls, JS steps, all of it — before downstream jobs skip on empty `pr_number`. The existing `# Note: We don't filter by event type here` comment was written for `workflow_run`, which is already scoped. `issue_comment` fires on every bot comment, human reply, and CI status post. With `cancel-in-progress: false`, these queue up rather than cancel.

Fix: add a job-level condition to `get-context`:
```yaml
get-context:
  if: github.event_name != 'issue_comment' || contains(github.event.comment.body, '/review')
```
This is the standard pattern for slash-command workflows and matches what the validation step does anyway — just moves the cheap text match before the expensive API calls.

---

**`check-label` step runs with empty `PR_NUMBER` for invalid `/review` commands** (line 360):

When validation fails (`valid=false`), `get-pr` returns an empty `pr_number`. The `check-label` block at line 360 still matches `EVENT_NAME = "issue_comment"`, attempts `gh pr edit  --remove-label ...` (fails silently due to `|| true`), sets `already_passed=false`, and exits 0. Downstream jobs presumably gate on `pr_number != ''`, so no real review runs — but this is a logic error that could cause confusion if that assumption breaks.

Fix: gate on `valid`:
```bash
if [ "$EVENT_NAME" = "issue_comment" ] && [ "${{ steps.review-command.outputs.valid }}" = "true" ]; then
```

### Worth Considering

The `run_id` fallback at line 167 (`|| echo "${{ github.run_id }}"`) sets `run_id` to the current ai-code-review workflow's own run ID if `gh run list` returns nothing. If downstream consumers use `run_id` to fetch PR Tests artifacts, they'd query the wrong run on that edge case.

### Conclusion

⚠️ Needs changes — both issues are straightforward one-liners.

**Comment by NickBorgers** [comment]
## Test Review

### Blocking Issues

**`/review` command is broken as written** — the PR-vs-issue guard at line 119 will never work correctly:

```bash
if [ -z "${{ github.event.issue.pull_request }}" ]; then
```

`github.event.issue.pull_request` is a JSON **object** when the comment is on a PR. GitHub Actions renders top-level object expressions as an empty string in `run:` contexts, so this condition is always true — every `issue_comment` event exits with `valid=false` regardless of whether it is on a PR or plain issue. The `/review` command will silently no-op every time.

Fix: use a scalar field from the object:
```bash
if [ -z "${{ github.event.issue.pull_request.url }}" ]; then
```
`pull_request.url` is a non-empty string for PR comments and empty/null for plain issue comments.

### Worth Considering

- **Concurrency group mismatch**: `workflow_run` events key on branch name (`ai-review-{branch}`); `issue_comment` events key on PR number (`ai-review-pr-{N}`). These are different groups, so a `/review` comment while an automatic review is still in progress won't be blocked — both complete and post duplicate review comments.

- **`run_id` fallback is the wrong run** (line 167): if `gh run list` returns nothing, `run_id` falls back to `${{ github.run_id }}` — the AI Code Review run ID, not a PR Tests run ID. Affects only the `fix-test-failures` path when `gh run list` fails.

### Conclusion
⚠️ Needs changes — the PR detection guard is broken and the `/review` command will not work without the `.url` fix.

**Comment by NickBorgers** [comment]
## 🚦 Merge Decision

**Decision: 🟢 GO**

**Findings addressed:**

- **Fixed** (Test Review blocking): PR-vs-issue detection guard used \`\${{ github.event.issue.pull_request }}\` (a JSON object), which GitHub Actions always renders as empty string in bash — so every \`/review\` command silently exited with \`valid=false\`. Changed to \`\${{ github.event.issue.pull_request.url }}\` (a scalar field) which correctly distinguishes PR comments from plain issue comments.

- **Fixed** (Code Review blocking): No job-level guard on \`get-context\` meant every comment on every issue/PR triggered checkout, API calls, and JS steps before downstream jobs could skip on empty \`pr_number\`. Added \`if: github.event_name != 'issue_comment' || contains(github.event.comment.body, '/review')\` to the \`get-context\` job, matching the standard slash-command workflow pattern.

- **Fixed** (Design Review blocking / Code+Test Review worth-considering): \`run_id\` fallback used \`echo "\${{ github.run_id }}"\` (the AI Code Review workflow's own ID) when \`gh run list\` returned nothing. Changed fallback to \`echo ""\` — an honest empty-string sentinel; callers already handle absent run_id and won't silently query artifacts from the wrong run.

- **Fixed** (Docs Review blocking): Updated \`docs/operations/AI_GHA_PIPELINES.md\` — added \`issue_comment\` to the Triggers section, added a full \`/review\` command section explaining who can use it and what security checks apply, updated the workflow table, and updated the Review Skip Mechanism bullet to mention the \`/review\` command alongside PR reopens.

- **Accepted** (Code Review worth-considering — \`check-label\` step with empty \`pr_number\`): The \`check-existing-reviews\` step already has \`if: steps.get-pr.outputs.pr_number != ''\`. When \`valid=false\`, \`get-pr\` returns empty \`pr_number\` and the step is skipped entirely, so the \`gh pr edit\` with an empty PR number never runs. The concern is addressed by the existing step-level guard.

- **Accepted** (Test Review worth-considering — concurrency group mismatch): \`workflow_run\` events key on branch name; \`issue_comment\` events key on PR number. A \`/review\` comment while an automatic review is in-progress will produce two independent runs and potentially two review comment threads. This is a UX inconvenience, not a correctness or security issue, and fixing it would require restructuring the concurrency strategy. Acceptable for now.

- **Accepted** (Design Review worth-considering — expression style): The \`pull_request.url\` fix already addresses this; the guard is now explicit about checking a scalar rather than an object.

**Applied fixes:** Committed as \`f13cb1f\` — fix: address review findings for /review command trigger. Pushed to \`align/add-review-command\`. Automatic merge with main is clean (no conflicts).

Ready for merge to main and production deployment.
